sync/rpc/server.go

```diff
@@ -809,12 +825,14 @@ func (s *SyncAgentServer) SnapshotPurge(ctx context.Context, req *empty.Empty) (
 			}
 		}
 
-		snapshotsInfo, err = s.getSnapshotsInfo()
+		snapshotsInfo, markedRemoved, err := s.getSnapshotsInfo()
 		if err != nil {
 			return
 		}
 
 		// We're tracing up from each leaf to the root
+		var removed int
+		progress := [2]float32{0, float32(markedRemoved)}
```

Comment:
The progress calculation is a bit too complex. We don't need to calculate in operation level since there are at most 2 (or 3?) operations for each remove snapshot. So  two levels of progress is enough:
1. Upper level: removed / markedAsRemoved * 100
2. Lower level: for each part in level 1, divided by the real-time coalescing progress (if it's a coalescing).